src/script/components/Modals/DetailViewModal/DetailViewModalFooter.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {FC} from 'react';
+import Icon from 'Components/Icon';
+
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {t} from 'Util/LocalizerUtil';
+
+import {Conversation} from '../../../entity/Conversation';
+import {ContentMessage} from '../../../entity/message/ContentMessage';
+
+interface DetailViewModalFooterProps {
+  messageEntity: ContentMessage;
+  conversationEntity: Conversation;
+  onLikeClick: (conversation: Conversation, message: ContentMessage) => void;
+  onReplyClick: (conversation: Conversation, message: ContentMessage) => void;
+  onDownloadClick: (message: ContentMessage) => void;
+}
+
+const DetailViewModalFooter: FC<DetailViewModalFooterProps> = ({
+  messageEntity,
+  conversationEntity,
+  onLikeClick,
+  onReplyClick,
+  onDownloadClick,
+}) => {
```

Comment:
Not a fan of using FC.
To be future-proof, I would recommend just typing the props and getting rid of FC.

Reply:
You mean:
({
```suggestion
const DetailViewModalFooter: FC<DetailViewModalFooterProps> = ({
  messageEntity,
  conversationEntity,
  onLikeClick,
  onReplyClick,
  onDownloadClick,
}: {
messageEntity: MessageEntity,
...rest
})) => {
```

or 

```suggestion
const DetailViewModalFooter = ({
  messageEntity,
  conversationEntity,
  onLikeClick,
  onReplyClick,
  onDownloadClick,
}): DetailViewModalFooterProps => {
```

Reply:
And why you recommend to not use FC ?